pkg/operation/botanist/component/extensions/crds/crds.go

```diff
@@ -0,0 +1,123 @@
+// Copyright (c) 2021 SAP SE or an SAP affiliate company. All rights reserved. This file is licensed under the Apache Software License, v. 2 except as noted otherwise in the LICENSE file
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package crds
+
+import (
+	"context"
+	_ "embed"
+
+	"github.com/gardener/gardener/pkg/client/kubernetes"
+	"github.com/gardener/gardener/pkg/operation/botanist/component"
+	"github.com/gardener/gardener/pkg/utils/flow"
+)
+
+var (
+	//go:embed templates/crd-backupbucket.tpl.yaml
+	backupBucketCRD string
+	//go:embed templates/crd-backupentry.tpl.yaml
+	backupEntryCRD string
+	//go:embed templates/crd-bastion.tpl.yaml
+	bastionCRD string
+	//go:embed templates/crd-cluster.tpl.yaml
+	clusterCRD string
+	//go:embed templates/crd-containerruntime.tpl.yaml
+	containerRuntimeCRD string
+	//go:embed templates/crd-controlplane.tpl.yaml
+	controlPlaneCRD string
+	//go:embed templates/crd-dnsentry.tpl.yaml
+	dnsEntryCRD string
+	//go:embed templates/crd-dnsowner.tpl.yaml
+	dnsOwnerCRD string
+	//go:embed templates/crd-dnsprovider.tpl.yaml
+	dnsProviderCRD string
+	//go:embed templates/crd-dnsrecord.tpl.yaml
+	dnsRecordCRD string
+	//go:embed templates/crd-extension.tpl.yaml
+	extensionCRD string
+	//go:embed templates/crd-infrastructure.tpl.yaml
+	infrastructureCRD string
+	//go:embed templates/crd-managedresources.tpl.yaml
+	managedResourcesCRD string
+	//go:embed templates/crd-network.tpl.yaml
+	networkCRD string
+	//go:embed templates/crd-operatingsystemconfig.tpl.yaml
+	operatingSystemConfigCRD string
+	//go:embed templates/crd-worker.tpl.yaml
+	workerCRD string
+
+	resources []string
+)
```

Comment:
Q: Is it possible to reuse the approach with embed.FS to read all template files (instead of having a var for each template file). Ref https://github.com/gardener/gardener/blob/v1.32.0/extensions/pkg/controller/worker/machine_crds.go

Reply:
I have not done it this way on purpose:
- We do not need to render templates with values, just get the content of the file
- I don't like very much how any file matching `*.tpl.yaml` pattern in that folder is applied in the cluster. Yes, it is easier to add new resources, but I prefer to be more explicit what exactly is installed in a cluster.